tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -638,19 +638,38 @@ function configure_containerd() {
 
 	mkdir -p /etc/containerd/
 
-	if [ $use_containerd_drop_in_conf_file = "false" ] && [ -f "$containerd_conf_file" ]; then
-		# only backup in case drop-in files are not supported, and when doing the backup
-		# only do it if a backup doesn't already exist (don't override original)
-		cp -n "$containerd_conf_file" "$containerd_conf_file_backup"
-	fi
+	# To avoid problems when this pod gets re-created let's be a bit paranoiac
+	# and try to configure/backup the most recent configuration more-less
+	# atomically. This shouldn't be that needed for drop-in configuration
+	# but shouldn't harm either.
+	local tmp_config_file=$(mktemp)
+	( for i in {1..10}; do
+		local pre_config="$(cat "$containerd_conf_file")"
+		echo "$pre_config" > "$tmp_config_file"
+
+		for shim in "${shims[@]}"; do
+			configure_containerd_runtime "$1" "$shim" "$tmp_config_file"
+		done
+
+		if [ "$(cat "$containerd_conf_file")" == "$pre_config" ]; then
+			# Nobody modified the containerd file, write our changes in
+			# least dangerous way (no atomicity of this section...)
+			if [ $use_containerd_drop_in_conf_file = "false" ] && [ -f "$containerd_conf_file" ]; then
+				# only backup in case drop-in files are not supported, and when doing the backup
+				# only do it if a backup doesn't already exist (don't override original)
+				cp -n "$containerd_conf_file" "$containerd_conf_file_backup"
+			fi
+			mv -f "$tmp_config_file" "$containerd_conf_file"
+			exit 0
+		fi
+		sleep $(($RANDOM / 1000))
+	done ) || { echo "Failed to configure containerd in 10 iterations, is someone else modifying the config?"; exit 1; }
```

Comment:
Instead of ` echo ... ; exit 1`, please, use the `die` function.

Reply:
ack

Reply:
Hi @ldoktor ,

I arrived right in time to as one more change: `s/is someone else modifying the config?/is someone else modifying ${containerd_conf_file}?`